lib/pages/chat_profile_info/chat_profile_info.dart

```diff
@@ -61,28 +98,135 @@ class ChatProfileInfoController extends State<ChatProfileInfo> {
         );
   }
 
-  void goToProfileShared() {
-    if (widget.isDraftInfo || widget.roomId == null) return;
-    Navigator.of(context).push(
-      CupertinoPageRoute(
-        builder: (context) {
-          return ChatProfileInfoShared(
-            roomId: widget.roomId!,
-            closeRightColumn: widget.onBack,
-          );
+  List<ChatDetailsPageModel> profileSharedPages() => profileSharedPagesList.map(
+        (page) {
+          switch (page) {
+            case ChatDetailsPage.media:
+              return ChatDetailsPageModel(
+                page: page,
+                child: _mediaListController == null
+                    ? const SizedBox()
+                    : ChatDetailsMediaPage(
+                        key: const PageStorageKey(
```

Comment:
i think you can create a mixin for this. because its is a method

Reply:
updated